pubsub/src/main/java/com/example/pubsub/PubSubExample.kt

```diff
@@ -0,0 +1,189 @@
+            // create a pull subscription with default acknowledgement deadline (= 10 seconds)
+            subscriptionAdminClient.createSubscription(subscriptionName, topicName, PushConfig.getDefaultInstance(), 0)
+        }
+        println("Subscription ${subscriptionName.project}:${subscriptionName.subscription} successfully created.")
+    } catch (e: ApiException) {
+        // example : code = ALREADY_EXISTS(409) implies subscription already exists
+        print(e.statusCode.code)
+        print(e.isRetryable)
+    }
+}
+
+private fun publishMsg(vararg args: String) { // expects 2 args: <topic> and <count>
+
+    if (args.size < 2) {
+        error("Bad input: command 'sub' expects 2 arguments. \n $usage")
+    }
+    val topicId = args[0]
+    val messageCount = args[1].toInt()
+    val topicName = ProjectTopicName.of(projectId, topicId)
+    lateinit var publisher: Publisher
+    val futures: MutableList<ApiFuture<String>> = mutableListOf()
+
+    try {
+        // Create a publisher instance with default settings bound to the topic
+        publisher = Publisher.newBuilder(topicName).build()
+
+        for (i in 1..messageCount) {
+            val message = "message-$i"
+
+            // convert message to bytes
+            val data = ByteString.copyFromUtf8(message)
+            val pubsubMessage = PubsubMessage.newBuilder()
+                    .setData(data)
+                    .build()
+
+            // Schedule a message to be published. Messages are automatically batched.
+            val future = publisher.publish(pubsubMessage)
+            futures.add(future)
+        }
+    } finally {
+        // Wait on any pending requests
+        val messageIds = ApiFutures.allAsList(futures).get()
+        messageIds.forEach { messageId -> println(messageId) }
+        publisher.shutdown()
+    }
+}
+
+private fun listenToSub(vararg args: String) { // expects 1 arg: <subscription> to listen to
+
+    if (args.isEmpty()) {
+        error("Bad input: command 'listen' expects 1 argument. \n $usage")
+    }
+
+    val messages = LinkedBlockingDeque<PubsubMessage>()
+
+    class MessageReceiverExample : MessageReceiver {
+        override fun receiveMessage(message: PubsubMessage, consumer: AckReplyConsumer) {
+            messages.offer(message)
+            consumer.ack()
+        }
+    }
+
+    // retrieve subscriber id, eg. my-sub
+    val subscriptionId = args[0]
+    val subscriptionName = ProjectSubscriptionName.of(projectId, subscriptionId)
+    lateinit var subscriber: Subscriber
+    try {
+        // create a subscriber bound to the asynchronous message receiver
+        subscriber = Subscriber.newBuilder(subscriptionName, MessageReceiverExample()).build()
+        subscriber.startAsync().awaitRunning()
+        // Continue to listen to messages
+        while (true) {
+            val message = messages.take()
+            println("Message Id: ${message.messageId}")
```

Comment:
Wouldn't mind having a message here, "Listening for messages...` to make it clear what's happening.